### PR TITLE
Encrypt credentials stored in peers table

### PR DESCRIPTION
**Pull request by iskakaushik**
This commit enhances security by encrypting peer options in the 'peers' table using AES encryption. AES was chosen for its reliability and widespread adoption in securing sensitive data. Changes include:

- Added `enc_key_id` column to the `peers` table.
- Modified `LoadPeer` and `CreatePeerNoValidate` functions to handle encryption/decryption.
- Updated environment configuration to support encryption key management.

### CODE REVIEW

File: flow/peerdbenv/config.go [hunk 1]
```diff
@@ -92,3 +92,17 @@ func PeerDBAlertingEmailSenderRegion() string {
 func PeerDBAlertingEmailSenderReplyToAddresses() string {
 	return GetEnvString("PEERDB_ALERTING_EMAIL_SENDER_REPLY_TO_ADDRESSES", "")
 }
+
+func PeerDBCurrentEncKeyID() string {
+	return GetEnvString("PEERDB_CURRENT_ENC_KEY_ID", "")
+}
+
+func PeerDBEncKeys() shared.PeerDBEncKeys {
+	return GetEnvJSON("PEERDB_ENC_KEYS", make(shared.PeerDBEncKeys, 0))
```
Comment (serprex) [inline flow/peerdbenv/config.go:101]:
```suggestion
	return GetEnvJSON[shared.PeerDBEncKeys]("PEERDB_ENC_KEYS", nil)
```

Comment (iskakaushik) [reply flow/peerdbenv/config.go:101]: i liked it before because i could invoke `.Get` on it without worrying about NPE.

File: flow/shared/enc.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+	if key == nil {
+		return nil, errors.New("encryption key is nil")
+	}
+
+	decodedKey, err := base64.StdEncoding.DecodeString(key.Value)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode base64 key: %w", err)
+	}
+
+	block, err := aes.NewCipher(decodedKey)
```
Comment (serprex) [inline flow/shared/enc.go:47]: should error if decodedKey not 32 bytes

**Comment by serprex** [comment]
Need to also add decrypting to `getPostgresPeerConfigs`. Could revise that code to loop over peer names, but that'd have to make N queries instead of just 1

File: flow/peerdbenv/config.go (hunk 1 above)
Comment (serprex) [reply flow/peerdbenv/config.go:101]: right, was thinking in terms of `map`

File: flow/connectors/core.go [hunk 3]
```diff
@@ -264,13 +265,19 @@ func LoadPeerType(ctx context.Context, catalogPool *pgxpool.Pool, peerName strin
 
 func LoadPeer(ctx context.Context, catalogPool *pgxpool.Pool, peerName string) (*protos.Peer, error) {
 	row := catalogPool.QueryRow(ctx, `
-		SELECT type, options
+		SELECT type, options, enc_key_id
```
Comment (heavycrystal) [inline flow/connectors/core.go:268]: is there a migration for this

File: flow/connectors/core.go [hunk 4]
```diff
@@ -264,13 +265,19 @@ func LoadPeerType(ctx context.Context, catalogPool *pgxpool.Pool, peerName strin
...
-	if err := row.Scan(&peer.Type, &peerOptions); err != nil {
+	var encKeyID string
+	var encPeerOptions []byte
+	if err := row.Scan(&peer.Type, &encPeerOptions, &encKeyID); err != nil {
+		return nil, fmt.Errorf("failed to load peer: %w", err)
+	}
+
+	peerOptions, err := decryptPeerOptions(encKeyID, encPeerOptions)
+	if err != nil {
 		return nil, fmt.Errorf("failed to load peer: %w", err)
```
Comment (heavycrystal) [inline flow/connectors/core.go:281]: error should be swallowed at some point

File: flow/connectors/core.go (hunk 3 above)
Comment (iskakaushik) [reply flow/connectors/core.go:268]: see: https://github.com/PeerDB-io/peerdb/pull/1883

File: flow/connectors/core.go (hunk 4 above)
Comment (serprex) [reply flow/connectors/core.go:281]: ?

Comment (serprex) [reply flow/connectors/core.go:281]: talked with Kevin, he's suggesting crypto related error messages shouldn't display to user, can review crypto error messages for leaks later

**Review by serprex — APPROVED** [review]

**Review by heavycrystal — APPROVED** [review]

**Pull request merged**